velox/dwio/dwrf/reader/BinaryStreamReader.cpp

```diff
@@ -111,7 +111,10 @@ BinaryStreamReader::getStatistics() const {
         "Corrupted file detected, Footer stats are missing, but stripes are present");
     for (auto node = 0; node < typesSize; node++) {
       if (columnSelector_.shouldReadNode(node)) {
-        stats[node] = proto::ColumnStatistics();
+        auto cs =
+            google::protobuf::Arena::CreateMessage<proto::ColumnStatistics>(
+                stripeReaderBase_.getReader().arena());
+        stats.emplace(node, ColumnStatisticsWrapper(cs));
```

Comment:
We still have the same problem, I think for this function you should keep it as it is, and create another one returning  `proto::orc::ColumnStatistics` if it is required by ORC as well.  Wrappers can only be created on call sites.

Reply:
@Yuhta I looked at the code again, ORC does not use this method. I'll keep the previous code logic. Thank you for testing this.